llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -399,6 +399,21 @@ adderGenerator(const int32_t From, const int32_t To, const int32_t StepSize) {
   };
 }
 
+// Move to the next generator if it is exhausted allowing to chain generators
+std::function<std::optional<int32_t>()> concatGenerators(
```

Comment:
Didn't you cerate a nice type alias for `std::function<std::optional<int32_t>()>`? Maybe you can re-use it?

Reply:
It used to have it, but it got lost in the see of git rebases. I modified it back again, thanks :)